### PR TITLE
Change event request flow

### PR DESCRIPTION
**Pull request by Namiii**
Implemented the new Request builder flow for Event request

### CODE REVIEW

File: src/test/java/co/omise/live/LiveEventRequestTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    @Test
+    @Ignore("only hit the network when we need to.")
+    public void testLiveEventListGet() throws Exception {
+        Client client = getLiveClient();
+
+        Request<ScopedList<Event>> request = new Event.ListRequestBuilder().build();
+
+
```
Comment (nuxzero) [inline src/test/java/co/omise/live/LiveEventRequestTest.java]: Could you remove a newline?

File: src/test/java/co/omise/live/LiveEventRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        Client client = getLiveClient();
+
+        Request<ScopedList<Event>> request =
+                new Event.ListRequestBuilder()
+                        .options(new ScopedList.Options()
+                                .limit(5)
+                                .order(Ordering.ReverseChronological))
+                        .build();
+
+
```
Comment (nuxzero) [inline src/test/java/co/omise/live/LiveEventRequestTest.java]: Could you remove a newline?

File: src/test/java/co/omise/live/LiveEventRequestTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        ScopedList<Event> events = client.sendRequest(request, new TypeReference<ScopedList<Event>>() {
+        });
+
+        assertEquals(5, events.getLimit());
+        assertEquals(5, events.getData().size());
+        assertEquals(Ordering.ReverseChronological, events.getOrder());
+        assertEquals(114, events.getTotal()); // This can easily break as you add events,
```
Comment (nuxzero) [inline src/test/java/co/omise/live/LiveEventRequestTest.java]: Do we have an option to do assertion without breaking here? Or check `events.getTotal() > 0`?

File: src/test/java/co/omise/live/LiveEventRequestTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import static org.junit.Assert.*;
+
+public class LiveEventRequestTest extends BaseLiveTest {
+
+    private static final String EVENT_ID = "YOUR_EVENT";
+
+    @Test
+    @Ignore("only hit when test on live.")
+    public void testGetEventSuccess() throws Exception {
+        Client client = getLiveClient();
```
Comment (nuxzero) [inline src/test/java/co/omise/live/LiveEventRequestTest.java]: How about move `client` to the global variable?

File: src/test/java/co/omise/live/LiveEventRequestTest.java (hunk 1 above)
Comment (Namiii) [reply src/test/java/co/omise/live/LiveEventRequestTest.java]: 👍

File: src/test/java/co/omise/live/LiveEventRequestTest.java (hunk 2 above)
Comment (Namiii) [reply src/test/java/co/omise/live/LiveEventRequestTest.java]: 👍

File: src/test/java/co/omise/live/LiveEventRequestTest.java (hunk 3 above)
Comment (Namiii) [reply src/test/java/co/omise/live/LiveEventRequestTest.java]: `0` would also be breakable in certain cases, but it's more reliable than a concrete number, so I'll change it to that.

File: src/test/java/co/omise/live/LiveEventRequestTest.java (hunk 4 above)
Comment (Namiii) [reply src/test/java/co/omise/live/LiveEventRequestTest.java]: Will do 👍

Comment (pitiphong-p) [reply src/test/java/co/omise/live/LiveEventRequestTest.java]: I think leave it like this is OK except we want to test on the state sharing part

Comment (Namiii) [reply src/test/java/co/omise/live/LiveEventRequestTest.java]: It's way too many calls to get the `Client` though, don't think there's any negative for sharing one client, is there?

Comment (pitiphong-p) [reply src/test/java/co/omise/live/LiveEventRequestTest.java]: My concern is the state sharing. Also the those client should be destroyed on test end anyway?

Comment (Namiii) [reply src/test/java/co/omise/live/LiveEventRequestTest.java]: I'm not really sure if there's much state to be shared in the client, I get your concern about the `Client` not being fresh when the test happens though. It's not supplied by DI, so I think it gets destroyed as soon as the tests are done anyway (as it's no longer in use)

Comment (nuxzero) [reply src/test/java/co/omise/live/LiveEventRequestTest.java]: Basically, the `setup()` will call for each test cases. I don't think we gonna have any problem with sharing state.

Comment (Namiii) [reply src/test/java/co/omise/live/LiveEventRequestTest.java]: Actually, he's right, it gets called for each test 🤦‍♂️

**Comment by Namiii** [comment]
@nuxzero changes done

**Pull request merged**